### PR TITLE
Allow themes to use any styles in the theme.json whether or not the block supports it

### PR DESCRIPTION
**Pull request by oandregal**
Implements https://github.com/WordPress/gutenberg/issues/29778#issuecomment-800887310

With this PR, block supports are no longer used to decide whether the styles in the theme.json for a block should be used or not. Block supports are still used to show UI controls in both the block & global styles sidebar (themes can use any style property, users can use use the style properties "supported" by the block).

Changes in this PR:

- Server:
  - Before: block supports were used to decide whether a particular theme.json style was allowed or not.
  - After: block supports are no longer used and **any** theme.json style property that exists can be used in any block (see test instructions).

- Client:
  - Before: block supports were used to decide whether a particular theme.json style was allowed or not + to create UI controls in the block and global sidebars.
  - After: block supports are only used to create UI controls in the block and global sidebars.

- Update theme.json docs accordingly.

## How to test

1. Make sure existing themes work as expected: using TT1-blocks go to the site editor and load the front-end. Verify they work as expected.

2. Make sure themes can use any style property in the theme.json file:

- Create a post / page with a paragraph. Publish.
- Go to the theme.json of the TT1-blocks theme and add this piece of code in the styles section:

```json
"core/paragraph": {
  "typography": {
    "fontFamily": "var(--wp--preset--font-family--cambria-georgia)"
  }
}
```

- Load the front-end and verify the paragraphs get the cambria / georgia family assigned.
- Load the editors and verify the paragraphs get the cambria / georgia familiy assigned. There shouldn't be any font family controls in either the block sidebar or the global sidebar.

Before this change, the font-family for the paragraph will be ignored.

### CODE REVIEW

**Comment by pbking** [comment]
It works and it's brilliant and I'm walking on sunshine.
This advances theme.json configuration SO MUCH!

**Comment by gziolo** [comment]
@nosolosw, can you provide a more detailed description of what has changed with this PR? I'm having a hard time understanding how it impacts a single block. In particular, I'd be interested in seeing how it impacts `block.json` and `theme.json` in practice.

**Comment by oandregal** [comment]
@gziolo expanded a bit on the "changes in this PR" section, does that help?

In terms of block.json, nothing changes. In terms of theme.json, the noticeable change is what I shared in the test instructions: that themes can know use style properties in the theme.json in blocks that don't declare support for them (ex: use a font-family for paragraphs despite the block doesn't declare support for font families).

**Comment by gziolo** [comment]
Thanks, @nosolosw. I think I get it now 😄

Code changes look good, I would appreciate some validation from other folks that know better how it works in practice.

**Review by gziolo — APPROVED** [review]

**Comment by mtias** [comment]
I think this is not only the right direction but also simplifies the codebase quite a bit

**Comment by gziolo** [comment]
@mtias, and the efforts required to keep the documentation up to date, instead we have this really nice sentence:

> Themes can use any style property via the `experimental-theme.json` for any block ― it's the theme's responsibility to verify that it works properly according the block markup, etc.

File: docs/how-to-guides/themes/theme-json.md [hunk 1]
```diff
@@ -315,7 +315,7 @@ Note that, the name of the variable is created by adding `--` in between each ne
 
 ### Styles
 
-Each block declares which style properties it exposes via the [block supports mechanism](../block-api/block-supports.md). The support declarations are used to automatically generate the UI controls for the block in the editor, as well as being available through the `experimental-theme.json` file for themes to target.
+Each block declares which style properties it exposes via the [block supports mechanism](../block-api/block-supports.md). The support declarations are used to automatically generate the UI controls for the block in the editor. Themes can use any style property via the `experimental-theme.json` for any block ― it's the theme's responsibility to verify that it works properly according the block markup, etc.
```
Comment (gziolo) [inline docs/how-to-guides/themes/theme-json.md]:
```suggestion
Each block declares which style properties it exposes via the [block supports mechanism](../block-api/block-supports.md). The support declarations are used to automatically generate the UI controls for the block in the editor. Themes can use any style property via the `experimental-theme.json` for any block ― it's the theme's responsibility to verify that it works properly according to the block markup, etc.
```

**Review by aristath — APPROVED** [review]
I love it, a lot simpler than what we had before and makes perfect sense to go in this direction. 👍

**Pull request merged**